src/factor.h

```diff
@@ -62,11 +73,19 @@ class CompactFactor {
    */
   inline double potential(const VariableInFactor vifs[],
                           const variable_value_t var_values[],
-                          const variable_id_t vid,
-                          const variable_value_t proposal) const {
+                          const variable_id_t vid = -1,
+                          const variable_value_t proposal = -1,
```

Comment:
Sorry, one last set of requests: Can we move these params to the end? and use `INVALID_ID` and `INVALID_VALUE`?